pulsar-client-tools/src/test/java/org/apache/pulsar/admin/cli/TestCmdSources.java

```diff
@@ -412,6 +412,15 @@ public void testBatchSourceConfigMissingDiscoveryTriggererClassName() throws Exc
         expectedSourceConfig.setBatchSourceConfig(batchSourceConfig);
         testCmdSourceConfigFile(testSourceConfig, expectedSourceConfig);
     }
+
+    @Test(expectedExceptions = ParameterException.class, expectedExceptionsMessageRegExp = "Invalid source type 'foo' " +
```

Comment:
could you add a happy path test ?

Reply:
I tried it:

```java

    @Test
    public void testCmdSourceConfigFileSourceType() throws Exception {
        ConnectorDefinition mockDefinition = new ConnectorDefinition();
        mockDefinition.setName("builtin");
        when(source.getBuiltInSources()).thenReturn(List.of(mockDefinition));

        SourceConfig sourceConfig = getSourceConfig();
        sourceConfig.setSourceType("builtin");

        SourceConfig exceptedConfig = getSourceConfig();
        exceptedConfig.setSourceType("builtin");
        exceptedConfig.setArchive("builtin://builtin");

        testCmdSourceConfigFile(sourceConfig, exceptedConfig);
    }
    
```

... while due to some dirty code inherit local run from cmd source the tests are hard to tune. That said, `validateSourceType` is inherited by `LocalSourceRunner` with total different manner.